### ISSUE TITLE
Lint error due to user type in UserSettings.tsx

### ISSUE THREAD

**Issue by mcapodici** [nicoalbanese/kirimase#66]
The code generated for `UserSettings.tsx` is

```
export default function UserSettings({
  user,
}: {
  user: { name?: string; email?: string };
}) {
  return (
    <>
      <UpdateNameCard name={user.name ?? ""} />
      <UpdateEmailCard email={user.email ?? ""} />
    </>
  );
}
```

This shows in the linter when referenced in `page.tsx` as

```
Type '{ name?: string | null | undefined; email?: string | null | undefined; image?: string | null | undefined; } & { id: string; }' is not assignable to type '{ name?: string | undefined; email?: string | undefined; }'.
  Types of property 'name' are incompatible.
    Type 'string | null | undefined' is not assignable to type 'string | undefined'.
      Type 'null' is not assignable to type 'string | undefined'.
```

I propose to change it to the line in `UserSettings.tsx`:

```
--  user: { name?: string; email?: string };
++  user: Session["user"];
```

Requires this import:

```
import { Session } from "next-auth";
```

**Comment by sangrepura** [comment]
Also, there is another minor mismatch for the `apiVersion`
```
src/lib/stripe/index.ts:4:3 - error TS2322: Type '"2023-08-16"' is not assignable to type '"2023-10-16"'.
4   apiVersion: "2023-08-16",
```
fix:
```
--- a/src/lib/stripe/index.ts
+++ b/src/lib/stripe/index.ts
-  apiVersion: "2023-08-16",
+  apiVersion: "2023-10-16",
```

**Comment by nicoalbanese** [comment]
Hey @mcapodici - thanks for this! This has been resolved with 3508b58 and e191319 and will be pushed to production later today 😊
